include/gz/math/Cone.hh

```diff
@@ -0,0 +1,192 @@
+/*
+ * Copyright 2024 CogniPilot Foundation
+ * Copyright 2024 Open Source Robotics Foundation
+ * Copyright 2024 Rudis Laboratories
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+#ifndef GZ_MATH_CONE_HH_
+#define GZ_MATH_CONE_HH_
+
+#include <optional>
+#include "gz/math/MassMatrix3.hh"
+#include "gz/math/Material.hh"
+#include "gz/math/Quaternion.hh"
+
+namespace gz
+{
+  namespace math
+  {
+    // Foward declarations
+    class ConePrivate;
+
+    // Inline bracket to help doxygen filtering.
+    inline namespace GZ_MATH_VERSION_NAMESPACE {
+    //
+    /// \class Cone Cone.hh gz/math/Cone.hh
+    /// \brief A representation of a cone.
+    ///
+    /// The cone class supports defining a cone with a radius,
+    /// length, rotational offset, and material properties. Radius and
+    /// length are in meters. See Material for more on material properties.
+    /// By default, a cone's length is aligned with the Z axis. The
```

Comment:
can you clarify the direction of the cone along the Z axis? Is the base or the vertex in the positive Z direction?

Reply:
Done.